src/backend/pipeline/cont_scheduler.c

```diff
@@ -409,6 +423,10 @@ cq_bgproc_main(Datum arg)
 		break;
 	case Worker:
 		am_cont_worker = true;
+
+		/* XXX - put this somewhere nicer */
+		AdhocMgrDeleteAdhocs();
```

Comment:
The function needs to access and modify the pipeline_query, which needs to happen per database.